core/src/test/java/nl/nn/adapterframework/scheduler/job/SendMessageJobTest.java

```diff
@@ -0,0 +1,57 @@
+package nl.nn.adapterframework.scheduler.job;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import nl.nn.adapterframework.core.SenderException;
+import nl.nn.adapterframework.core.TimeoutException;
+import nl.nn.adapterframework.senders.IbisLocalSender;
+import nl.nn.adapterframework.stream.Message;
+
+class SendMessageJobTest {
+
+	private final SendMessageJob jobDef = new SendMessageJob();
+	private final IbisLocalSender localSenderMock = mock(IbisLocalSender.class);
+
+	@BeforeEach
+	void setup(){
+		jobDef.setMessage("messageMock");
+	}
+
+	@Test
+	void testHappyExecuteCall() throws JobExecutionException, TimeoutException, SenderException, IOException {
```

Comment:
In unit tests hanteer ik tegenwoordig altijd "A A A" testing: expliciet aangeven van "Arrange", "Act", en "Assert".

In de test zet ik bovenaan een comment `// Arrange`, voor de regel(s) waar de te testen code wordt aangeroepen zet ik een comment `// Act`, en daarna wanneer ik de assertions ga doen dan een comment `// Assert`.

Dan is het sneller duidelijk waar in de test je moet kijken voor het uitvoeren van de eigenlijke code, etc.

Wanneer nodig dan herhaal ik de comments met bijvoorbeeld `// Act 2`, `// Assert 2`, etc.

Soms, bijvoorbeeld bij `assertThrows`, zijn "Act" en "Assert" in 1. Dan zet ik er een comment bij `// Act / Assert`.

Reply:
Dat kende ik nog niet. De structuur die ik meestal aanhoud is:
1.  een lege regel tussen de validatie en uitvoer vd method
2. een beschrijvende test-methode naam kiezen, waaruit blijkt waarom de test bestaat
3. bij meerdere unit tests de setup zaken die overlappen naar een @BeforeEach blok verplaatsen
4. comments toevoegen als het groter wordt en niet direct duidelijk wat er gebeurt. Uiteraard is dat goed om te toetsen door een reviewer. 

Ik zal eens kijken of ik jouw methodiek kan toepassen bij grotere unit tests een volgende keer. Oke?

Reply:
Beschrijvende method naam een gebruik van `setup` methods zijn natuurlijk altijd wenselijk!

De comments `// Arrange`, `// Act`, `// Assert` zijn extra, om in de bestaande test code het onderscheid aan te geven.

Makkelijk toe te voegen zonder de tests te herschrijven, en maken de tests leesbaarder omdat je sneller de test-code kan scannen voor de comments. :-)

Reply:
Oke thanks voor de tips! Ik zal het toepassen als de testen groter worden. :-)